app/src/main/java/com/mapbox/maps/testapp/examples/SecondaryDisplayPresentationActivity.kt

```diff
@@ -0,0 +1,204 @@
+import androidx.core.content.ContextCompat
+import androidx.core.view.children
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import com.mapbox.maps.MapView
+import com.mapbox.maps.testapp.R
+import com.mapbox.maps.testapp.databinding.ActivitySecondaryDisplayPresentationBinding
+import com.mapbox.maps.testapp.databinding.ItemDisplayInfoBinding
+
+class SecondaryDisplayPresentationActivity : AppCompatActivity() {
+  private lateinit var displayManager: DisplayManager
+  private var currentPresentation: Presentation? = null
+  private lateinit var binding: ActivitySecondaryDisplayPresentationBinding
+  private val onDismissListener = DialogInterface.OnDismissListener { dialog ->
+    if (dialog == currentPresentation) {
+      Log.i(TAG, "Presentation was dismissed.")
+      currentPresentation = null
+      updateDisplaysInfo()
+    }
+  }
+  private val displayListener = object : DisplayManager.DisplayListener {
+    override fun onDisplayAdded(displayId: Int) {
+      Log.d(TAG, "Display #$displayId added.")
+      updateDisplaysInfo()
+    }
+
+    override fun onDisplayRemoved(displayId: Int) {
+      Log.d(TAG, "Display #$displayId removed.")
+      updateDisplaysInfo()
+    }
+
+    override fun onDisplayChanged(displayId: Int) {
+      Log.d(TAG, "Display #$displayId changed.")
+      updateDisplaysInfo()
+    }
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    binding = ActivitySecondaryDisplayPresentationBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+    displayManager = ContextCompat.getSystemService(this, DisplayManager::class.java)!!
+  }
+
+  override fun onResume() {
+    super.onResume()
+    updateDisplaysInfo()
+    // Register to receive events from the display manager.
+    displayManager.registerDisplayListener(displayListener, null)
+  }
+
+  override fun onPause() {
+    // Unregister from the display manager.
+    displayManager.unregisterDisplayListener(displayListener)
+    super.onPause()
+  }
+
+  override fun onStop() {
+    // Dismiss the presentation when the activity is not visible.
+    currentPresentation?.let {
+      Log.i(TAG, "Dismissing presentation because the activity is no longer visible.")
+      it.dismiss()
+      currentPresentation = null
+    }
+    super.onStop()
+  }
+
+  private fun updateDisplaysInfo() {
+    clearDisplaysInfo()
+
+    val thisActivityDisplay = windowManager.defaultDisplay
+    val displays = displayManager.displays
+    val secondaryDisplays = displays.filterNot { it.displayId == thisActivityDisplay.displayId }
+    if (secondaryDisplays.isEmpty()) {
+      binding.displayNone.visibility = View.VISIBLE
+    } else {
+      secondaryDisplays.forEach { display ->
+        Log.d(TAG, "updatePresentation: Found display $display")
+        val displayInfoBinding =
```

Comment:
could maybe use something like this to reduce repeated use of displayInfoBinding? Just my personal preference.
```
ItemDisplayInfoBinding.inflate(
            LayoutInflater.from(binding.root.context),
            binding.root,
            true
          ).apply {
             root.tag = DISPLAY_INFO_TAG
             displayInfo.text = "${display.name} [${display.displayId}]"
             displayBt.isEnabled = true
           // etc
```